versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5775 @@
+
+A `Pair` can be declared with comma-separated values in parentheses (`()`). The components of a pair value are accessed using its `left` and `right` accessors.
+
+```wdl
+Pair[Int, Array[String]] data = (5, ["hello", "goodbye"])
+Int i = p.left
+String hello = data.right[0]
+```
+
+##### Map[P, Y]
+
+A `Map` represents an associative array of key-value pairs. All of the keys must be of the same (primitive) type, and all of the values must be of the same type, but keys and values can be different types.
+
+A `Map` can be declared with comma-separated key-value pairs in braces (`{}`). Key-value pairs are delimited by `:`. The value of a specific key can be accessed by placing the key in brackets after the declaration name. Accessing a non-existent key of a `Map` results in an error.
+
+```wdl
+Map[Int, Int] int_to_int = {1: 10, 2: 11}
+Map[String, Int] string_to_int = { "a": 1, "b": 2 }
+Map[File, Array[Int]] file_to_ints = {
+  "/path/to/file1": [0, 1, 2],
+  "/path/to/file2": [9, 8, 7]
+}
+
+string_to_int["b"]  # => 2
+string_to_int["c"]  # error!  
+```
+
+A `Map` is insertion ordered, meaning the order in which elements are added to the `Map` is preserved, for example when [✨ converting a `Map` to an array of `Pair`s](#-arraypairp-y-as_pairsmapp-y).
+
+```wdl
+Map[Int, String] int_to_string = { 2: "hello", 1: "goodbye" }
+as_pairs(int_to_string)  # => [(2, "hello"), (1, "goodbye")]
+```
+
+##### 🗑 Object
+
+An `Object` is an unordered associative array of identifier-value pairs, where values may be of any type and are not specified explicitly.
+
+An `Object` can be declared using syntax similar to a map literal, but with two differences:
+
+1. The `object` keyword comes before the opening brace (`{`).
+2. The identifiers are not quoted.
+
+The value of a specific member of an object value can be accessed by placing a `.` followed by the member identifier after the object declaration name.
+
+```wdl
+Object f = object {
+  a: 10,
+  b: 11
+}
+Int i = f.a
+```
+
+Due to the lack of explicitness in the typing of `Object` being at odds with the goal of being able to know the type information of all WDL declarations, **the `Object` type, the `object` literal syntax, and all of the [standard library functions](#-object-read_objectstringfile) with `Object` parameters or return values have been deprecated and will be removed in the next major version of the WDL specification**.
+
+##### Custom Types (Structs)
+
+WDL provides the ability to define custom compound types called [`struct`s](#struct-definition). Structs are defined directly in the WDL document and are usable like any other type. 
+
+A `struct` is declared using the `struct` keyword, followed by a unique name, followed by member declarations within braces. A declaration with a custom type can be assigned a value using struct literal syntax, which is identical to the [objects literal syntax](#-object) with the exception that the `struct` name is used instead of the `object` keyword.
```

Comment:
Do you want to be referring here to the objects syntax given that it will be deprecated?  You'll have to fix this later.  Perhaps now is a good time to put whatever text you were hoping the readers to read here, HERE, even if it's slightly duplicated in the short term.

Reply:
Maybe throw in a trash can icon?

Reply:
Yes, given that the object literal syntax has already been discussed. In the 2.0 spec when objects have been removed, we'll also remove references to objects and object literal syntax throughout the spec.

Reply:
Actually, this is a good point. I think it makes better sense to talk about custom types first and objects second.